video/decode/vd_lavc.c

```diff
@@ -1235,6 +1236,12 @@ static int receive_frame(struct mp_filter *vd, struct mp_frame *out_frame)
     int ret = decode_frame(vd);
 
     if (ctx->hwdec_failed) {
+        // If software fallback is disabled, immediately EOF here.
+        if (ctx->opts->software_fallback == INT_MAX) {
+            MP_WARN(ctx, "Software decoding fallback is disabled.\n");
+            return AVERROR_EOF;
+        }
+
```

Comment:
This isn't right now, as the fallback loop below may go through other hwdecs that could succeed. I think this logic will need to go inside force_fallback to have it skip the software fallback. Which is more correct anyway, as that's where the decision gets made, regardless of how we trigger the fallback.

Reply:
Oh right, it's different now. Would it work simply after the `force_fallback` loop? I don't think it's possible to reach that point after all the hwdecs have been tried?

Reply:
I went ahead and moved it there which I believe happens after everything in the loop is tried.